### PR TITLE
Report benchmark performance per Pull Request

### PR DESCRIPTION
**Pull request by li-boxuan**
This commit adds github-action-benchmark which
can report performance change between current
commit and previous commit.

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

**Comment by li-boxuan** [comment]
Example: https://github.com/li-boxuan/janusgraph/pull/33#commitcomment-73542256

File: .github/workflows/ci-benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          key: ${{ runner.os }}-benchmark
+
+      # Run `github-action-benchmark` action
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          tool: 'customSmallerIsBetter'
+          output-file-path: janusgraph-benchmark/benchmark.json
+          external-data-json-path: ./cache/benchmark-data.json
+          fail-on-alert: true
```
Comment (farodin91) [inline .github/workflows/ci-benchmark.yml:84]: Do we want to activate the failure after sometime?

**Comment by li-boxuan** [comment]
According to https://github.com/actions/first-interaction/issues/10, unfortunately, we cannot have `github-actions` bot to leave a comment on PR due to security restrictions.

**Comment by li-boxuan** [comment]
@farodin91 Do you want to review it again?

**Review by farodin91 — APPROVED** [review]
LGTM! Thank you @li-boxuan.

**Pull request merged**